2_base_and_sensor.ipynb

```diff
@@ -436,135 +812,259 @@
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "outputs": [],
    "source": [
-    "# scan_data.angle_maxの値をラジアンからdegに変換します\n",
+    "# scan_data.angle_maxの値をラジアンからdegに変換します Convert value of scan_data.angle_max from radians to deg\n",
     "scan_data.angle_max / math.pi * 180  # math.pi = π"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "レーザスキャナのデータ本体は、`ranges`という名前の配列に格納されており、配列の長さは721のようです。"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Actual laser scanner data is stored in the array named `ranges`, and the length of the array seems to be 721:"
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "outputs": [],
    "source": [
-    "# データの配列の長さを習得\n",
+    "# データの配列の長さを取得 Get length of the array\n",
     "len(scan_data.ranges)"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "スキャン範囲（-120度から120度）上の各観測点がデータ化されているので、配列の真ん中（=361番目）の要素が「ロボット正面から壁までの距離」を表します。\n",
     "\n",
     "センサ値の単位はメートルです。"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Since each observation point on the scan range (-120 degrees to 120 degrees) is digitized from left to right, the element in the middle of the array (= 361th) represents the \"distance from the front of the robot to the wall\".\n",
+    "\n",
+    "The unit of sensor value is meters."
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "outputs": [],
    "source": [
-    "# 361番目のデータにアクセス\n",
+    "# 361番目のデータにアクセス Access the 361th data\n",
     "scan_data.ranges[360]"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "# センサ値を使ってロボットを制御する"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "# Control the robot using sensor values"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "センサ値を使ってロボットを制御してみましょう。"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Let's control the robot using the sensor value."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "壁の1メートル手前で止まる（壁の1メートル手前まで前進する）プログラムを書いてみましょう。\n",
     "\n",
     "while文を使うと以下のように書けると思います。"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Let's write a program that stops 1 meter before the wall (go straight until the robot reaches 1meter before the wall).\n",
```

Comment:
Changed "1meter" to "1 meter".
Keep reading, there's more:

Use the following if there is a specific wall that you had referred to before:

    "Let's write a program that stops 1 meter before the wall (go straight until the robot reaches 1 meter before the wall).\n",

Use the following if you have not introduced a specific wall, yet:

    "Let's write a program that stops 1 meter before the wall (go straight until the robot reaches 1 meter before the wall).\n",